torch_em/data/datasets/histopathology/consep.py

```diff
@@ -0,0 +1,268 @@
+        block_seg = segmentation[bb]
+        block_segs.append(block_seg)
+
+    seg_ids = np.unique(segmentation)
+    rag = compute_rag(segmentation)
+    edge_disaffinities = np.full(rag.numberOfEdges, 0.9, dtype="float32")
+
+    for block_id in tqdm(range(n_blocks), desc="Stitch tiles with overlap"):
+        for axis in range(ndim):
+            ngb_id = blocking.getNeighborId(block_id, axis, lower=True)
+            if ngb_id == -1:
+                continue
+
+            this_block = blocking.getBlockWithHalo(block_id, list(halo))
+            ngb_block = blocking.getBlockWithHalo(ngb_id, list(halo))
+
+            this_seg, ngb_seg = block_segs[block_id], block_segs[ngb_id]
+
+            face = tuple(
+                slice(beg_out, end_out) if d != axis else slice(beg_out, beg_in + halo[d])
+                for d, (beg_out, end_out, beg_in) in enumerate(
+                    zip(this_block.outerBlock.begin, this_block.outerBlock.end, this_block.innerBlock.begin)
+                )
+            )
+
+            this_face_bb = tuple(
+                slice(fa.start - offset, fa.stop - offset) for fa, offset in zip(face, this_block.outerBlock.begin)
+            )
+            ngb_face_bb = tuple(
+                slice(fa.start - offset, fa.stop - offset) for fa, offset in zip(face, ngb_block.outerBlock.begin)
+            )
+            this_face = this_seg[this_face_bb]
+            ngb_face = ngb_seg[ngb_face_bb]
+            assert this_face.shape == ngb_face.shape, (this_face.shape, ngb_face.shape)
+
+            # COMMENT: I visualized the faces, seems like they look as expected.
+
+            # NOTE: I have a feeling that the overlap method is either not working for here or I am missing something.
+            overlap_comp = overlap(this_face, ngb_face)
+            this_ids = np.unique(this_face)
+            overlaps = {this_id: overlap_comp.overlapArraysNormalized(this_id, sorted=False) for this_id in this_ids}
+            overlap_ids = {this_id: ovlps[0] for this_id, ovlps in overlaps.items()}
+            overlap_values = {this_id: ovlps[1] for this_id, ovlps in overlaps.items()}
+            overlap_uv_ids = np.array([
+                [this_id, ovlp_id] for this_id, ovlp_ids in overlap_ids.items() for ovlp_id in ovlp_ids
+            ])
+            overlap_values = np.array([ovlp for ovlps in overlap_values.values() for ovlp in ovlps], dtype="float32")
+            assert len(overlap_uv_ids) == len(overlap_values)
+
+            valid_uv_ids = np.isin(overlap_uv_ids, seg_ids).all(axis=1)
+            if valid_uv_ids.sum() == 0:
+                continue
+            overlap_uv_ids, overlap_values = overlap_uv_ids[valid_uv_ids], overlap_values[valid_uv_ids]
+            assert len(overlap_uv_ids) == len(overlap_values)
+
+            edge_ids = rag.findEdges(overlap_uv_ids)
+            # NOTE: All edges I receive seem to be '-1' (something is wrong with computing 'overlap_uv_ids')
```

Comment:
This all looks correct to me. If the issue persists after fixing the problem with the ids we need to debug this with a breakpoint together.